frontend/src/components/shared/Dropdown/Selectbox.tsx

```diff
@@ -49,6 +51,32 @@ interface SelectOption {
   value: string
 }
 
+// Add a custom filterOptions method to filter options only based on labels.
+// The baseweb default method filters based on labels or indeces
+// More details: https://github.com/streamlit/streamlit/issues/1010
+// Also filters using fuzzy search powered by fzy.js. Automatically handles
+// upper/lowercase.
+export function fuzzyFilterSelectOptions(
+  options: SelectOption[],
+  pattern: string
+): readonly SelectOption[] {
+  // wrap options in lodash-chainable array
+  let arr = _(options)
+
+  // keep the options that fuzzy-match our pattern
+  arr = arr.filter((opt: SelectOption) => fzy.hasMatch(pattern, opt.label))
+
+  // if the user provided a pattern, sort by score, highest first; otherwise,
+  // retain original order (scoring based on empty pattern makes no sense)
+  if (pattern) {
+    arr = arr.sortBy((opt: SelectOption) => fzy.score(pattern, opt.label))
```

Comment:
nit: No strong opinion on this, but it might be cleaner to just chain the lodash calls without reassigning `arr`.

Something like
```javascript
if (!pattern) {
    return options
}

return _(options)
    .filter(...)
    .sortBy(...)
    .reverse()
    .values()
```

is equivalent and avoids variable reassignment

Reply:
I had this at first, but the `.sortBy()` and `.reverse()` are conditional on `pattern`, and I don't know how to add a conditional into a chain. 

I could chain the `sortBy` and `reverse()` together though.

Reply:
Never mind, saw your comment above about putting the `.filter()` in the `if` too. That cleans things up.